pkg/services/ngalert/schedule/schedule_test.go

```diff
@@ -290,3 +368,79 @@ func concatenate(keys []models.AlertRuleKey) string {
 	}
 	return fmt.Sprintf("[%s]", strings.Join(s, ","))
 }
+
+// fakeNotifier represents a fake internal Alertmanager.
+type fakeNotifier struct{}
+
+func (n *fakeNotifier) PutAlerts(alerts apimodels.PostableAlerts) error {
+	return nil
+}
+
+type fakeExternalAlertmanager struct {
+	t      *testing.T
+	mtx    sync.Mutex
+	alerts amv2.PostableAlerts
+	server *httptest.Server
+}
+
+func newFakeExternalAlertmanager(t *testing.T) *fakeExternalAlertmanager {
+	t.Helper()
+
+	am := &fakeExternalAlertmanager{
+		t:      t,
+		alerts: amv2.PostableAlerts{},
+	}
+	am.server = httptest.NewServer(http.HandlerFunc(am.Handler()))
+
+	return am
+}
+
+func (am *fakeExternalAlertmanager) AlertNamesCompare(expected []string) bool {
+	n := []string{}
+	alerts := am.Alerts()
+
+	if len(expected) != len(alerts) {
+		return false
+	}
+
+	for _, a := range am.Alerts() {
+		for k, v := range a.Alert.Labels {
+			if k == model.AlertNameLabel {
+				n = append(n, v)
+			}
+		}
+	}
+
+	return cmp.Equal(expected, n)
```

Comment:
`assert` has `assert.ObjectsAreEqual` method we could use here instead of importing another package.